lsp-mode.el

```diff
@@ -3686,47 +3691,50 @@ RENDER-ALL - nil if only the signature should be rendered."
 
 (defun lsp-hover ()
   "Display signature or hover info based on the current position."
-  (if (and lsp--hover-saved-bounds
-           (lsp--point-in-bounds-p lsp--hover-saved-bounds))
-      (lsp--eldoc-message lsp--eldoc-saved-message)
-
-    (setq lsp--hover-saved-bounds nil
-          lsp--eldoc-saved-message nil)
-    (let ((request-id (cl-incf lsp-hover-request-id)) (pending 0))
-      (when (and lsp-eldoc-enable-hover (lsp--capability "hoverProvider"))
-        (cl-incf pending)
-        (lsp-request-async
-         "textDocument/hover"
-         (lsp--text-document-position-params)
-         (lambda (hover)
-           (when (and (eq request-id lsp-hover-request-id))
-             (when hover
-               (when-let (range (gethash "range" hover))
-                 (setq lsp--hover-saved-bounds (lsp--range-to-region range)))
-               (-let (((&hash "contents") hover))
-                 (when-let (message
-                            (and contents (lsp--render-on-hover-content contents lsp-eldoc-render-all)))
-                   (when (or (and (not lsp-eldoc-prefer-signature-help) (setq pending 1))
+  (let* ((whitespace-or-newline (looking-at "[[:space:]\n]"))
+         (skip-hover (and lsp-hover-skip-whitespace-or-newline whitespace-or-newline)))
+    (unless skip-hover
```

Comment:
what happens if you move from identifier to `space`? I think that it will preserve the original value so you should have else leg and call lsp--eldoc-message with an empty message.